packages/engine/src/scene/systems/ShadowSystem.tsx

```diff
@@ -266,16 +273,17 @@ const EntityChildCSMReactor = (props: { entity: Entity; rendererEntity: Entity }
 }
 
 function _CSMReactor() {
-  const renderSettingsEntity = useEntityContext()
-  const rendererEntity = useScene(renderSettingsEntity)
+  const rendererEntity = useEntityContext()
 
   if (!rendererEntity) return null
+  const renderSettingsEntity = useChildWithComponent(rendererEntity, RenderSettingsComponent)
```

Comment:
the same number of hooks must always exist in a reactor, move the early escape after all hooks